doc/pages/options.asciidoc

```diff
@@ -139,13 +139,16 @@ are exclusively available to built-in options.
     `set -remove` removes the given specs from the list. +
 
 *completions*::
-    a list of `<text>|<select cmd>|<menu text>` candidates,
+    a list of `<text>[|<filter text>]|<select cmd>|<menu text>` candidates,
     except for the first element which follows the
     `<line>.<column>[+<length>]@<timestamp>` format to define where the
     completion apply in the buffer.
     Any `|` or `\` characters that occur within the `<text>`,
-    `<select cmd>`, or `<menu text>` fields should be escaped as `\|`
-    or `\\`.
+    `<filter text>`, `<select cmd>`, or `<menu text>` fields should be
+    escaped as `\|` or `\\`.
+
+    Kakoune will filter away candidates that do not match what the user types.
+    Candidates are matched against the optional `<filter text>`, or `<text>`.
```

Comment:
yep, I added a commit that tries to explain this

> - the completion menu displays "filter text" (if provided) and/or "text"

Currently that's an "or", but at least for the Julia completion of Unicode symbols it would definitely work well to display `\xrat (℞)` here.

I did not yet find other examples where the text-to-insert is different from the filter text.
If the filter text can be very similar to the text-to-insert, it may not be wise to display both.

Reply:
The more general form of completions is "type the name of something, get something else". It could be something shorter (like typing "Smiling Face with Open Mouth and Tightly-Closed Eyes" to get 😆), or it could be something much longer (like typing "defprop" to get a snippet that defines a property access method — I know @alexherbo2 has been working on something like this).

If I had to pick a convention for people to follow, it would probably be that the completion menu itself displays the text that I need to type to narrow down the list of completions, while the menu text (if short) or `info -menu` (if long) is used to preview the thing that will be inserted when I accept an option.

Reply:
Right, clients that use filter-text can use the menu-text and the on-select handler to display what would be inserted.